### ISSUE TITLE
[Android] Trying to view Spotify in desktop mode in Brave causes an infinite loop

### ISSUE THREAD

**Issue by Brave-Matt** [brave/brave-browser#42571]
### Description

When going to open.spotify.com in the browser on Android, the site loads just fine. However, if you try to view this site in desktop mode, the site will infinitely load/reload/refresh itself. This appears to happen with Shields on/off.

### Steps to reproduce

1. In Brave on Android visit open.spotify.com
2. Open the main menu and check the `Desktop view` box
3. Observe the site stuck in an infinite loop

### Actual result

Infinite loop

### Expected result

Site should load normally in desktop view.

### Reproduces how often

Easily reproduced

### Brave version

v1.73.91

### Device

- Brand/model: Samsung Galaxy (likely others)
- Android version: 13/14

### Channel information

- [x] release (stable)
- [x] beta
- [x] nightly

### Reproducibility

- [ ] with Brave Shields disabled
- [ ] with Brave Rewards disabled
- [ ] in the latest version of Chrome

### Miscellaneous information

User report:
https://community.brave.com/t/desktop-site-mode-not-working-for-https-open-spotify-com-the-page-goes-in-infinite-loop-forever/583116

**Comment by Al3676** [comment]
Latest version Chrome high value iam wishes

**Comment by bsclifton** [comment]
is this related to a recent issue we had solved? cc: @atuchin-m

**Comment by atuchin-m** [comment]
Nope, it's definitely another thing. The know issue was an OOM crash. Looks like a webcompat problem.
